chainio/types.go

```diff
@@ -0,0 +1,37 @@
+package chainio
+
+import (
+	"encoding/hex"
+	"errors"
+	"strings"
+)
+
+const (
+	// We use uint8 to count the number of quorums, so we can have at most 255 quorums,
+	// which means the max ID can not be larger than 254 (from 0 to 254, there are 255
+	// different IDs).
+	MaxQuorumID = 254
+)
+
+type QuorumID = uint8
+
+type OperatorID = [32]byte
```

Comment:
I feel like these types (and helper methods) should belong somewhere outside than chainio. Why do we want to redefine them outside existing core package?

Reply:
Where should they be defined? Currently, `core/v2` depends on `chainio` for operator state types, and `chainio` is meant to depend only on chain bindings. 

We could potentially reverse the ordering, but thi is inconvenient because it means core can use the chainio objects. Any other ideas?

Reply:
I see. Conceptually, I thought it made sense for chainio to depend on core, but I guess the argument can be made the other way too. I'm fine either way. 
It's bit confusing having the same types (`OperatorID`, `QuorumID`) defined again here though, in addition to the existing `core`. Is having this duplicate type perferred over reusing `core.OperatorID`?

Reply:
These are simply type aliases, not type definitions. I'm using this just as a tool for maintaining easy consistency and upgradeability within this package. From that perspective, I wonder if it is less confusing? 

I'd prefer not to have any dependencies on `core` as this will be deprecated.